chalk-solve/src/lib.rs

```diff
@@ -85,6 +88,9 @@ pub trait RustIrDatabase<I: Interner>: Debug {
     fn well_known_trait_id(&self, well_known_trait: WellKnownTrait) -> Option<TraitId<I>>;
 
     fn interner(&self) -> &I;
+
+    /// Returns the ids of all auto traits.
+    fn auto_traits(&self) -> Vec<TraitId<I>>;
```

Comment:
Hmm, I didn't notice this before. I don't think we can land with this method... we don't want to require clients to enumerate all traits that exist.